src/components/dropdownMenu.js

```diff
@@ -0,0 +1,210 @@
+    if (showDropdownMenu) {
+      document.body.addEventListener('click', handleOutsideClick, true);
+      document.body.addEventListener('keydown', handleKeyboardEvent, true);
+      scrollContainer.addEventListener('scroll', determinePosition, true);
+    }
+    return () => {
+      if (showDropdownMenu) {
+        document.body.removeEventListener('click', handleOutsideClick, true);
+        document.body.removeEventListener('keydown', handleKeyboardEvent, true);
+        scrollContainer.removeEventListener('scroll', determinePosition, true);
+      }
+    };
+  });
+
+  const toggleDropdown = useCallback((toggleState) => {
+    setShowDropdownMenu(toggleState);
+    if (!toggleState) {
+      setKeySelected(null);
+      setPositionTop(false);
+    }
+  }, []);
+
+  const handleKeyboardEvent = useCallback((e) => {
+    e.preventDefault();
+    const len = options.length;
+
+    switch (e.keyCode) {
+      case UP_ARROW:
+        const i = keySelected < 1 ? len - 1 : (keySelected - 1) % len;
+        setKeySelected(i);
+        return;
+      case DOWN_ARROW:
+        const index = keySelected !== null ? ((keySelected + 1) % len) : 0;
+        setKeySelected(index);
+        return;
+      case ENTER_KEY:
+        if (showDropdownMenu && onSelect) onSelect(options[keySelected], e);
+        if (!showDropdownMenu) {
+          dropdownButton.current.focus();
+        } else {
+          dropdownButton.current.blur();
+        }
+        toggleDropdown(!showDropdownMenu);
+        return;
+      case TAB_KEY:
+        toggleDropdown(!showDropdownMenu);
+    }
+  }, [showDropdownMenu, options, keySelected]);
+
+  const determinePosition = useCallback(() => {
+    const [target, content] = componentRef.current.children;
+    const windowHeight = window.innerHeight;
+    const { top, height } = content.getBoundingClientRect();
+    const { height: contentHeight } = target.getBoundingClientRect();
+    const totalHeight = top + height;
+
+    if ((totalHeight > windowHeight) && (top > height)) {
+      setPositionTop(`${contentHeight}px`);
+    } else {
+      setPositionTop(false);
+    }
+  }, [componentRef]);
+
+  const handleOutsideClick = useCallback((e) => {
+    if (componentRef.current.contains(e.target)) {
+      return;
+    }
+    toggleDropdown(false);
+  }, [componentRef]);
+
+  const dropdownMenus = options
+    ? options.map((option, index) => (
+      <Menu
+        onClick={e => {
+          toggleDropdown(!showDropdownMenu);
+          onSelect && onSelect(option, e);
+        }}
+        key={option.value}
+        onMouseOver={() => setKeySelected(index)}
+        hover={keySelected === index}
+        variant={variant}
+      >
+        {option.label}
+      </Menu>
+    ))
+    : noOptionMessage;
+
+  return (
+    <DropdownContainer ref={componentRef}>
+      <DropDown
+        variant={variant}
+        onClick={() => toggleDropdown(!showDropdownMenu)}
+        id={id}
+        tabIndex='0'
```

Comment:
- [ ] any reason for hardcoding tabIndex? What if the developer wants to override the tabIndex